client/src/resources/transaction.rs

```diff
@@ -99,6 +236,16 @@ impl Transaction {
     pub fn fee_meta_xdr(&self) -> &String {
         &self.fee_meta_xdr
     }
+
+    /// The value/contents of the memo for this transaction
+    pub fn memo(&self) -> Option<&MemoValue>{
```

Comment:
I'd rather this just return `Option<&Memo>` as an enum type with the values in the middle.